Lib/shutil.py

```diff
@@ -550,7 +557,10 @@ def move(src, dst, copy_function=copy2):
             os.rename(src, dst)
             return
 
-        real_dst = os.path.join(dst, _basename(src))
+        # Using _basename instead of os.path.basename is important, as we must
+        # ignore any trailing slash to avoid the basename returning ''
```

Comment:
Can you explain the need for `str()` too here? (It's clear from the PR title, but it won't be clear immediately to future readers of the code.)

Reply:
done!

Reply:
> Ah, the problem is that os.fspath(x) returns bytes if x is bytes, which breaks _basename. Oh well.

Actually rstrip breaks on bytes anyways, so this isn't a breaking change. Let's get this in already. :)